scripts/faq_script.ts

```diff
@@ -0,0 +1,168 @@
+import { DateTime } from "luxon";
+import { Readable } from "node:stream";
+
+import logger from "@adonisjs/core/services/logger";
+
+import GuideArticle from "#models/guide_article";
+import GuideQuestion from "#models/guide_question";
+import FilesService from "#services/files_service";
+
+interface GuideArticlesOld {
+  data: {
+    id: number;
+    name: string;
+    cover: string;
+    short_description: string;
+    description: string | null;
+    order: number;
+    questions: number[];
+  }[];
+}
+
+interface GuideQuestionsOld {
+  data: {
+    id: number;
+    status: string;
+    date_created: string;
+    date_updated: string;
+    question: string;
+    answer: string;
+    type: number;
+  }[];
+}
+
+interface PivotTable {
+  data: {
+    id: number;
+    FAQ_Types_id: number | null;
+    FAQ_id: number;
+    sort: number | null;
+  }[];
+}
+
+async function uploadImage(imageUrl: string) {
+  const filesService = new FilesService();
+
+  const fetchedImage = await fetch(imageUrl);
+  if (!fetchedImage.ok) {
+    throw new Error(
+      `Failed to download image. HTTP status: ${fetchedImage.status}`,
+    );
+  }
+
+  const extension = fetchedImage.headers.get("content-type")?.split("/")[1];
```

Comment:
my concern about the way the file extension is derived still wasn't addressed.
I've found out that there's an endpoint for file metadata in directus and I myself just extract the extension from the on-disk filename that directus reports.
The endpoint URL is `https://admin.topwr.solvro.pl/files/:id`.